cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -4189,6 +4197,102 @@ mod tests {
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                r#"expected an entity uid, found literal `"view_actions"`"#
+            ).help(
+                "try including the entity type if you intended this string to be an entity identifier"
+            ).exactly_one_underline(r#""view_actions""#).build());
+        });
+    }
+
+    #[test]
+    fn scope_compare_to_name() {
+        let p_src = r#"permit(principal == User, action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                "expected an entity uid or matching template slot, found name `User`"
+            ).help(
+                "try using `is` to test for an entity type"
+            ).exactly_one_underline("User").build());
+        });
+        let p_src = r#"permit(principal in Group, action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                "expected an entity uid or matching template slot, found name `Group`"
+            ).help(
+                "try using `is` to test for an entity type"
+            ).exactly_one_underline("Group").build());
+        });
+        let p_src = r#"permit(principal, action, resource in Album);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                "expected an entity uid or matching template slot, found name `Album`"
+            ).help(
+                "try using `is` to test for an entity type"
+            ).exactly_one_underline("Album").build());
+        });
+        // Testing for absence of help message because actions scope doesn't support `is`.
+        let p_src = r#"permit(principal, action == Action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                "expected an entity uid, found name `Action`"
+            ).exactly_one_underline("Action").build());
+        });
+    }
+
+    #[test]
+    fn scope_and() {
+        let p_src = r#"permit(principal == User::"alice" && principal in Group::"jane_friends", action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                "expected an entity uid or matching template slot, found a `&&` expression"
+            ).help(
+                "The policy scope can only contain one constraint per variable. Consider moving the second operand of this `&&` into a `when` condition.",
```

Comment:
New help text is great! Just a picky comment about capitalization 🙂 
```suggestion
                "the policy scope can only contain one constraint per variable. Consider moving the second operand of this `&&` into a `when` condition",
```